src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random random = new Random();
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Color color = colorSupplier.getRandomColor();
+        FigureType figureType = getRandomFigureType();
+
+        return switch (figureType) {
+            case CIRCLE -> new Circle(color.name(), random.nextDouble() * 10 + 1);
+            case SQUARE -> new Square(color.name(), random.nextDouble() * 10 + 1);
+            case RECTANGLE -> new Rectangle(color.name(), random.nextDouble() * 10 + 1,
+                    random.nextDouble() * 10 + 1);
+            case RIGHT_TRIANGLE -> new RightTriangle(color.name(), random.nextDouble() * 10 + 1,
+                    random.nextDouble() * 10 + 1);
+            case ISOSCELES_TRAPEZOID -> new IsoscelesTrapezoid(color.name(),
+                    random.nextDouble() * 10 + 1,
+                    random.nextDouble() * 10 + 1, random.nextDouble() * 10 + 1);
+            default -> throw new IllegalStateException("Unexpected figure type: " + figureType);
```

Comment:
DO NOT re-request reviews if you didn't implemented comments, all those magic numbers should be CONSTANTS, also you can make a variable before `switch` where you will store random number and then use it in all cases.

`double someName = random.nextDouble(10) + 1` - but make those numbers CONSTANTS